components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -1520,7 +1520,8 @@ private UsersGetResponse filterUsersBySingleAttribute(ExpressionNode node, Map<S
                     node.getValue()));
         }
         // Check whether the filter operation is supported by the users endpoint.
-        if (isFilteringNotSupported(node.getOperation())) {
+        if (isFilteringNotSupported(node.getOperation()) &&
```

Comment:
The `and` condition may not required as `NE` operator is marked as supported operator.

Reply:
Resolved via https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/640/commits/4f00e65cbb2c0f0b0032e2998653b0871d542196